### ISSUE TITLE
How to adjust the table to fit container

### ISSUE THREAD

**Issue by tienkhoa16** [jsamr/react-native-render-html-table-bridge#17]
Hi,

May I ask how I can adjust the table to fit the container? I tried setting the fitContainer in cssRules to true but it seems to make no difference.

**Comment by jsamr** [comment]
@tienkhoa16 I will be happy to help you if you provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of your issue!

**Comment by tienkhoa16** [comment]
**For a big screen, the table is too small:**
![BigScreen](https://user-images.githubusercontent.com/60292763/86325459-a6dcc000-bc72-11ea-981f-5ad5bd29c292.png)


**For a smaller screen, the table is too big that user has to scroll left and right to see it:**
![SmallScreen](https://user-images.githubusercontent.com/60292763/86326255-01c2e700-bc74-11ea-90a2-8e4ecd177248.png)


**My code for rendering the screen:**
```
import React, {PureComponent} from 'react';
import {ScrollView} from 'react-native';
import HTML from 'react-native-render-html';
import { IGNORED_TAGS, alterNode, makeTableRenderer } from 'react-native-render-html-table-bridge';
import WebView from 'react-native-webview';

let html = `
<table>
    <thead>
    <tr>
        <th rowspan="2">S.No</th>
        <th rowspan="2">Date</th>
        <th colspan="3" align="center">AM</th>
        <th colspan="3" align="center">PM</th>
    </tr> 
    <tr>    
        <th>Temp <SUP>&deg;</SUP>C</th>
        <th >COVID sx*</th>
        <th >Same home sx*</th>														
        <th >Temp <SUP>&deg;</SUP>C</th>
        <th >COVID sx*</th>
        <th >Same home sx*</th>
    </tr>
    </thead>
    <tbody>	
            <tr> 
                <td >1</td>
                <td>08/06 Mon</td>
                    <td >
                        36.4
                        </td>
                        <td >                        
                        No
                        </td>
                        <td >                        
                        No
                    </td>                
                    <td >
                        36.9
                        </td>
                        <td >                        
                        No
                        </td>
                        <td >                        
                        No
                    </td>                    
            </tr>
            <tr> 
                <td >2</td>
                <td>07/06 Sun</td>
                    <td >
                        36.4
                        </td>
                        <td >                        
                        No
                        </td>
                        <td >                        
                        No
                    </td>
                    <td >
                        36.8
                        </td>
                        <td >                        
                        No
                        </td>
                        <td >                        
                        No
                    </td>
            </tr>
    </tbody>
</table>`

const config = {
    WebViewComponent: WebView
};

const renderers = {
  table: makeTableRenderer(config)
};

const htmlConfig = {
  alterNode,
  renderers,
  ignoredTags: IGNORED_TAGS
};

export default class Example extends PureComponent<Props> {
  render() {
    return (
        <ScrollView contentContainerStyle={{ paddingHorizontal: 10 }} style={{ marginTop: 50 }}>
        <HTML html={html} {...htmlConfig}/>
      </ScrollView>
    )
  }
}
```

In _css-rules.js_ file in the node_modules folder, I tried changing _fitContainer_ property in _defaultTableStylesSpecs_ from false to true but it makes no difference
```
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultTableStylesSpecs = {
    selectableText: false,
    fitContainer: true,     // by default it is false, but when I set it to true, it makes no difference
    cellPaddingEm: 0.25,
    borderWidthPx: 0.25,
    linkColor: '#3498DB',
    fontFamily: 'sans-serif',
    thBorderColor: '#3f5c7a',
    tdBorderColor: '#b5b5b5',
    thOddBackground: '#253546',
    thOddColor: '#FFFFFF',
    thEvenBackground: '#253546',
    thEvenColor: '#FFFFFF',
    trOddBackground: '#e2e2e2',
    trOddColor: '#333333',
    trEvenBackground: '#FFFFFF',
    trEvenColor: '#333333'
};
```

So how can I shrink the table content to fit my screen by default (like what react-native-webview does)? Thanks!

**Comment by jsamr** [comment]
@tienkhoa16 I'm working on it. Will keep you updated.

**Comment by jsamr** [comment]
@tienkhoa16 I've just released v0.6.1 with features that should help you. 

A few comments:

> how can I shrink the table content to fit my screen by default (like what react-native-webview does)

I'm not aware that react-native-webview does that. But if you want to play with RNWV props, you can use [`webViewProps` config attribute](https://github.com/jsamr/react-native-render-html-table-bridge#webviewprops). The web engine will try to fit the table into the container width, but if it can't, the table will overflow. The only solution is to have a smaller font and / or cell spacing. You can use `fontSizePx` and `cellSpacingEm` for that regard (see bellow).

>  For a big screen, the table is too small

I have fixed the issue by providing `fitContainerWidth` attribute to `tableStyleSpecs` config, see bellow.

``` js
import React, {PureComponent} from 'react';
import {ScrollView} from 'react-native';
import HTML from 'react-native-render-html';
import { IGNORED_TAGS, alterNode, makeTableRenderer } from 'react-native-render-html-table-bridge';
import WebView from 'react-native-webview';

let html = `
<table>
    <thead>
    <tr>
        <th rowspan="2">S.No</th>
        <th rowspan="2">Date</th>
        <th colspan="3" align="center">AM</th>
        <th colspan="3" align="center">PM</th>
    </tr> 
    <tr>    
        <th>Temp <SUP>&deg;</SUP>C</th>
        <th >COVID sx*</th>
        <th >Same home sx*</th>														
        <th >Temp <SUP>&deg;</SUP>C</th>
        <th >COVID sx*</th>
        <th >Same home sx*</th>
    </tr>
    </thead>
    <tbody>	
            <tr> 
                <td >1</td>
                <td>08/06 Mon</td>
                    <td >
                        36.4
                        </td>
                        <td >                        
                        No
                        </td>
                        <td >                        
                        No
                    </td>                
                    <td >
                        36.9
                        </td>
                        <td >                        
                        No
                        </td>
                        <td >                        
                        No
                    </td>                    
            </tr>
            <tr> 
                <td >2</td>
                <td>07/06 Sun</td>
                    <td >
                        36.4
                        </td>
                        <td >                        
                        No
                        </td>
                        <td >                        
                        No
                    </td>
                    <td >
                        36.8
                        </td>
                        <td >                        
                        No
                        </td>
                        <td >                        
                        No
                    </td>
            </tr>
    </tbody>
</table>`

const config = {
  WebViewComponent: WebView,
  tableStyleSpecs: {
    cellSpacingEm: 0.1,
    fitContainerWidth: true,
    fontSizePx: 14
  }
};

const renderers = {
  table: makeTableRenderer(config)
};

const htmlConfig = {
  alterNode,
  renderers,
  ignoredTags: IGNORED_TAGS
};

export default class Example extends PureComponent<Props> {
  render() {
    return (
      <ScrollView contentContainerStyle={{ paddingHorizontal: 10 }} style={{ marginTop: 50 }}>
        <HTML html={html} {...htmlConfig}/>
      </ScrollView>
    )
  }
}
```

Hope that helps

**Comment by tienkhoa16** [comment]
@jsamr Thanks a lot. It works as expected for me now 👍
